src/apps/bridge/sensor_drivers/seapointTurbiditySensor.cpp

```diff
@@ -0,0 +1,175 @@
+  (void)type;
+  (void)version;
+  printf("Seapoint Turbidity data received from node %016" PRIx64 ", on topic: %.*s\n", node_id,
+         topic_len, topic);
+  SeapointTurbidity_t *turbidity_sensor =
+      static_cast<SeapointTurbidity_t *>(sensorControllerFindSensorById(node_id));
+  if (turbidity_sensor && turbidity_sensor->type == SENSOR_TYPE_SEAPOINT_TURBIDITY) {
+    if (xSemaphoreTake(turbidity_sensor->_mutex, portMAX_DELAY)) {
+      static BmSeapointTurbidityDataMsg::Data turbidity_data;
+      if (BmSeapointTurbidityDataMsg::decode(turbidity_data, data, data_len) == CborNoError) {
+        char *log_buf = static_cast<char *>(pvPortMalloc(SENSOR_LOG_BUF_SIZE));
+        configASSERT(log_buf);
+        turbidity_sensor->turbidity_s_ftu.addSample(turbidity_data.s_signal);
+        turbidity_sensor->turbidity_r_ftu.addSample(turbidity_data.r_signal);
+        turbidity_sensor->reading_count++;
+        // Large floats get formatted in scientific notation,
+        // so we print integer seconds and millis separately.
+        uint64_t reading_time_sec = turbidity_data.header.reading_time_utc_ms / 1000U;
+        uint32_t reading_time_millis = turbidity_data.header.reading_time_utc_ms % 1000U;
+        uint64_t sensor_reading_time_sec = turbidity_data.header.sensor_reading_time_ms / 1000U;
+        uint32_t sensor_reading_time_millis =
+            turbidity_data.header.sensor_reading_time_ms % 1000U;
+
+        uint32_t current_timestamp = pdTICKS_TO_MS(xTaskGetTickCount());
+        if (current_timestamp - turbidity_sensor->last_timestamp >
+                DEFAULT_TURBIDITY_READING_PERIOD_MS + 1000U ||
+            turbidity_sensor->reading_count == 1U) {
+          printf("Updating Seapoint Turbidity %016" PRIx64
+                 " node position, current_time = %" PRIu32 ", last_time = %" PRIu32
+                 ", reading count: %" PRIu32 "\n",
+                 node_id, current_timestamp, turbidity_sensor->last_timestamp,
+                 turbidity_sensor->reading_count);
+          turbidity_sensor->node_position =
+              topology_sampler_get_node_position(node_id, pdTICKS_TO_MS(5000));
+        }
+        turbidity_sensor->last_timestamp = current_timestamp;
+
+        size_t log_buflen =
+            snprintf(log_buf, SENSOR_LOG_BUF_SIZE,
+                     "%016" PRIx64 ","     // Node Id
+                     "%" PRIi8 ","         // node_position
+                     "seapoint_turbidity," // node_app_name
+                     "%" PRIu64 ","        // reading_uptime_millis
+                     "%" PRIu64 "."        // reading_time_utc_ms seconds part
+                     "%03" PRIu32 ","      // reading_time_utc_ms millis part
+                     "%" PRIu64 "."        // sensor_reading_time_ms seconds part
+                     "%03" PRIu32 ","      // sensor_reading_time_ms millis part
+                     "%.4f,"               // s_signal
+                     "%.3f\n",             // r_signal
```

Comment:
Just checking/learning — is this the precision that actually comes from the sensor?

Reply:
For the s_signal I believe it has 4 decimal places. The r_signal may only have 1 (based on my testing).
Here are some example raw lines from the sensor (I got different values by moving my hand over it):
```
line: 0.0030,990.3
line: 4.6124,1084.5
line: 50.102,1290.8
line: 103.88,1413.2
line: 94.568,1398.3
line: 77.588,1359.9
line: 15.611,1163.3
line: 0.0054,992.8
```
The s_signal definitely goes down to 4 decimals at really close to 0 FTU. The R signal only ever seems to have 1. I cannot find in the data sheet an explicit number of decimals for either signal. 

There is this tid-bit in the data sheet:
> • Noise Floor: < 0.005 FTU RMS
It apparently applies to both s and r signals. I'm not sure if that means the the output will ever have 3 decimal places or not. I put three for the r_signal just in case for now and four for the s_signal since it is outputting that many.